mtp_api/apps/mtp_auth/serializers.py

```diff
@@ -2,14 +2,17 @@
 
 from rest_framework import serializers
 
+from .models import PrisonUserMapping
+
 
 class UserSerializer(serializers.ModelSerializer):
     prisons = serializers.SerializerMethodField()
 
     def get_prisons(self, obj):
-        return [
-            prison.pk for prison in obj.prisonusermapping.prisons.all()
-        ]
+        try:
+            return list(obj.prisonusermapping.prisons.values_list('pk', flat=True))
+        except PrisonUserMapping.DoesNotExist:
+            return []
```

Comment:
So it does. That wasn't entirely clear because it does them all in one test. It'd be much clearer as multiple tests, I think.